src/caches/beliefpropagationcache.jl

```diff
@@ -40,8 +48,18 @@ function BeliefPropagationCache(tn, partitioned_vertices; kwargs...)
   return BeliefPropagationCache(ptn; kwargs...)
 end
 
-function BeliefPropagationCache(tn; kwargs...)
-  return BeliefPropagationCache(tn, default_partitioning(tn); kwargs...)
+function BeliefPropagationCache(
+  tn; partitioned_vertices=default_partitioned_vertices(tn), kwargs...
+)
+  return BeliefPropagationCache(tn, partitioned_vertices; kwargs...)
+end
+
+function cache(
+  alg::Algorithm"bp",
+  tn;
+  cache_construction_kwargs=default_cache_construction_kwargs(alg, tn),
```

Comment:
I think it would be nicer to just use flattened keyword arguments here rather than a single keyword argument which accepts a NamedTuple of the keyword arguments.